File: src/fiefapp.svelte/src/models/building.ts
import type { Builder } from './builder';

interface IBuilding {
  id: string;
  builderId?: string;
  type: string;
  amount: number;
  woodworkThisYear: number;
  stoneworkThisYear: number;
  smithsworkThisYear: number;
  woodThisYear: number;
  stoneThisYear: number;
  ironThisYear: number;
  upkeep: number;
  woodwork: number;
  stonework: number;
  smithswork: number;
  wood: number;
  stone: number;
  iron: number;
}

export class Building implements IBuilding {
  id: string;
  builder?: Builder;
  amount: number;
  woodworkThisYear: number;
  stoneworkThisYear: number;
  smithsworkThisYear: number;
  woodThisYear: number;
  stoneThisYear: number;
  ironThisYear: number;
  type: string;
  upkeep: number;
  woodwork: number;
  stonework: number;
  smithswork: number;
  wood: number;
  stone: number;
  iron: number;

  constructor() {
    this.id = '';
    this.builder = undefined;
    this.amount = 0;
    this.woodworkThisYear = 0;
    this.stoneworkThisYear = 0;
    this.smithsworkThisYear = 0;
    this.woodThisYear = 0;
    this.stoneThisYear = 0;
    this.ironThisYear = 0;
    this.type = '';
    this.upkeep = 0;
    this.woodwork = 0;
    this.stonework = 0;
    this.smithswork = 0;
    this.wood = 0;
    this.stone = 0;
    this.iron = 0;
  }
}

File: src/fiefapp.entities/Building.cs
using fiefapp.entities.Interfaces;
using System;
using System.Collections.Generic;

namespace fiefapp.entities
{
    public class Building : IBaseEntity
    {
        public Guid Id { get; set; }
        public string Type { get; set; }
        public decimal Upkeep { get; set; }
        public int Stonework { get; set; }
        public int Woodwork { get; set; }
        public int Smithswork { get; set; }
        public int Stone { get; set; }
        public int Wood { get; set; }
        public int Iron { get; set; }
        public int Amount { get; set; }
        public int WoodworkThisYear { get; set; }
        public int StoneworkThisYear { get; set; }
        public int SmithsworkThisYear { get; set; }
        public int WoodThisYear { get; set; }
        public int StoneThisYear { get; set; }
        public int IronThisYear { get; set; }
        public Guid? BuilderId { get; set; }
        public Builder Builder { get; set; }
        public ICollection<Guid> GamesessionIds { get; set; }
        public ICollection<Gamesession> Gamesessions { get; set; }
    }
}
